cryptoservice/crypto_service.go

```diff
@@ -24,16 +25,16 @@ const (
 type CryptoService struct {
 	gun        string
 	passphrase string
-	keyStore   *trustmanager.KeyFileStore
+	keyStore   trustmanager.KeyStore
 }
 
 // NewCryptoService returns an instance of CryptoService
-func NewCryptoService(gun string, keyStore *trustmanager.KeyFileStore, passphrase string) *CryptoService {
+func NewCryptoService(gun string, keyStore trustmanager.KeyStore, passphrase string) *CryptoService {
 	return &CryptoService{gun: gun, keyStore: keyStore, passphrase: passphrase}
 }
 
 // Create is used to generate keys for targets, snapshots and timestamps
-func (ccs *CryptoService) Create(role string, algorithm data.KeyAlgorithm) (*data.PublicKey, error) {
+func (ccs *CryptoService) Create(role string, algorithm data.KeyAlgorithm) (data.Key, error) {
```

Comment:
I changed the CryptoService interface so the return types are `data.Key` (an interface) rather than `*data.PublicKey` (pointer to a structure). This makes it possible to implement the hardware signing as a cryptoservice, because it doesn't use the PublicKey type (it needs a PKCS11 object handle, for example).